cirq-core/cirq/sim/simulator_base.py

```diff
@@ -160,17 +176,29 @@ def _core_iterator(
         Yields:
             StepResults from simulating a Moment of the Circuit.
         """
+
+        def merge_states(sim_state: Dict['cirq.Qid', TActOnArgs]) -> TActOnArgs:
+            final_args = None
+            for args in set(sim_state.values()):
+                final_args = args if final_args is None else cast(TActOnArgs, final_args).join(args)
+            if final_args is None:
+                return self._create_act_on_arg(initial_state=0, qubits=[], logs={})
```

Comment:
Could this be simplified to:
```
final_args = self._create_act_on_arg(initial_state=0, qubits=[], logs={})
for args in set(sim_state.values()):
    final_args = final_args.join(args)
```
Or does this produce incorrect behavior / introduce significant overhead costs?